### PR TITLE
Update data-type-date-and-time.md

### PR DESCRIPTION
**Pull request by zoujia-cm**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.0 (TiDB 5.0 versions)
- [x] v4.0 (TiDB 4.0 versions)
- [x] v3.1 (TiDB 3.1 versions)
- [x] v3.0 (TiDB 3.0 versions)
- [x] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- The PR is translated from:

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by TomShawn** [comment]
@zoujia-cm Thanks for your contribution! Could you please sign the [Contributor License Agreement](https://cla-assistant.io/pingcap/docs-cn?pullRequest=5619) for this PR to be merged?

**Comment by zoujia-cm** [comment]
> @zoujia-cm Thanks for your contribution! Could you please sign the [Contributor License Agreement](https://cla-assistant.io/pingcap/docs-cn?pullRequest=5619) for this PR to be merged?

hi，I have signed the CLA

**Comment by zoujia-cm** [comment]
\cc @XuHuaiyu take a look please

File: data-type-date-and-time.md [hunk 1]
```diff
@@ -151,7 +151,7 @@ TIME[(fsp)]
 
 ### `DATETIME` 类型
 
-`DATETIME` 类型是日期和时间的组合，格式为 `YYYY-MM-DD HH:MM:SS[.fraction]`。支持的范围是 `1000-01-01 00:00:00.000000` 到 `9999-12-31 23:59:59.999999`。`fsp` 参数表示秒精度，取值范围为 0~6，默认值为 0。TiDB 支持字符串或数字转换为 `DATETIME` 类型。
+`DATETIME` 类型是日期和时间的组合，格式为 `YYYY-MM-DD HH:MM:SS[.fraction]`。支持的范围是 `0000-01-01 00:00:00.000000` 到 `9999-12-31 23:59:59.999999`。`fsp` 参数表示秒精度，取值范围为 0~6，默认值为 0。TiDB 支持字符串或数字转换为 `DATETIME` 类型。
```
Comment (XuHuaiyu) [inline data-type-date-and-time.md:154]: It's wrong.
According to https://dev.mysql.com/doc/refman/8.0/en/datetime.html

> The DATETIME type is used for values that contain both date and time parts. MySQL retrieves and displays DATETIME values in 'YYYY-MM-DD hh:mm:ss' format. The supported range is '1000-01-01 00:00:00' to '9999-12-31 23:59:59'.

Comment (zoujia-cm) [reply data-type-date-and-time.md:154]: > It's wrong.
> According to https://dev.mysql.com/doc/refman/8.0/en/datetime.html
> 
> > The DATETIME type is used for values that contain both date and time parts. MySQL retrieves and displays DATETIME values in 'YYYY-MM-DD hh:mm:ss' format. The supported range is '1000-01-01 00:00:00' to '9999-12-31 23:59:59'.

I have get the information from the website. But the fact is users can insert data range of 0000-01-01 00:00:00 to 9999-12-31 23:59:59 into table.

Comment (zoujia-cm) [reply data-type-date-and-time.md:154]: > It's wrong.
> According to https://dev.mysql.com/doc/refman/8.0/en/datetime.html
> 
> > The DATETIME type is used for values that contain both date and time parts. MySQL retrieves and displays DATETIME values in 'YYYY-MM-DD hh:mm:ss' format. The supported range is '1000-01-01 00:00:00' to '9999-12-31 23:59:59'.

I don't know how to post a image here. You can try it in TiDB. 
execute these sql:
  create table dtest(dt datetime);
  insert into dtest values('0000-01-01 00:00:00'),('999-01-01 11:22:33');
then
  select * from dtest;
you will find these data can be insert successfully.

**Comment by zoujia-cm** [comment]
@XuHuaiyu  pelese take a look

**Comment by XuHuaiyu** [comment]
`For the DATE and DATETIME range descriptions, “supported” means that although earlier values might work, there is no guarantee.`

https://dev.mysql.com/doc/refman/8.0/en/datetime.html

**Comment by zoujia-cm** [comment]
> `For the DATE and DATETIME range descriptions, “supported” means that although earlier values might work, there is no guarantee.`
> 
> https://dev.mysql.com/doc/refman/8.0/en/datetime.html

I do not think it is persuasive. From this point of view, I can say the range supports '9997-01-01 23:59:59' to '9998-01-02 00:00:00'. it is also right. But it is not the information that users want to get.  **Of course, this is just a personal opinion.**

**Comment by XuHuaiyu** [comment]
这里的意思是说 '1000-01-01' to '9999-12-31' 的范围，参与查询、计算等操作，确保正确性。
而在这个范围外的，不对正确性做保证

**Comment by TomShawn** [comment]
> 这里的意思是说 '1000-01-01' to '9999-12-31' 的范围，参与查询、计算等操作，确保正确性。
> 而在这个范围外的，不对正确性做保证

I agree with @XuHuaiyu. Maybe you can successfully insert `0000-01-01`, but we do not guarantee its correctness. For our user documentation, we only document the information whose correctness can be guaranteed by us, which is also backed up by MySQL docs.

**Comment by TomShawn** [comment]
@zoujia-cm To better manage the PRs in this repo, I'll close this one for now. If you have further updates, feel free to reopen it. Thanks!

**Pull request closed without merging**